File: Payflow SDK for .NET/SampleStoreVB/src/PayPal/Payments/SampleStore/VB/Handler/CreditHandler.vb
Option Explicit On 
Option Strict On

Imports PayPal.Payments.DataObjects
Imports PayPal.Payments.Transactions
Imports PayPal.Payments.SampleStore.VB.Common
Imports PayPal.Payments.Common.Utility

#Region "Copyright"

'Unpublished Proprietary Program Material
'This material is proprietary to PayPal, Inc. and is not to be reproduced, 
'used or disclosed except in accordance with a written license agreement 
'with PayPal, Inc.. 
'(C) Copyright 2005  PayPal, Inc.   All Rights Reserved.
'PayPal, Inc. believes that this material furnished herewith is accurate 
'and reliable.  However, no responsibility, financial or otherwise, can be 
' accepted for any consequences arising out of the use of this material. 

'The copyright notice above does not evidence any actual or intended 
'publication of such source code. 

#End Region

Namespace PayPal.Payments.SampleStore.VB.Handler
    ''' -----------------------------------------------------------------------------
    ''' <summary>
    ''' This class creates object of CreditTransaction class, call its 
    ''' SubmitTransaction method, gets the response object and returns the same
    ''' </summary>
    ''' <remarks>
    ''' </remarks>
    ''' -----------------------------------------------------------------------------

    Public Class CreditHandler
        Inherits BaseTransactionHandler
#Region "SubmitTransaction"
        ''' <summary>
        ''' This function overrides the submittransaction function of the base class.
        ''' This function creates object of CreditTransaction class, call its 
        ''' SubmitTransaction method, gets the response object and returns the same
        ''' </summary>
        ''' <remarks>
        ''' </remarks>
        ''' -----------------------------------------------------------------------------

        Public Overrides Function SubmitTransaction(ByVal RequestData As Hashtable) As Hashtable
            Try
                Dim CreditTrans As CreditTransaction
                Dim OrigId, RequestId As String
                Dim Count As Integer
                Dim ResponseTable As New Hashtable

                RequestId = RequestData.Item(SampleStoreConstants.FIELD_REQUESTID).ToString
                OrigId = RequestData.Item(SampleStoreConstants.FIELD_ORIGID).ToString

                If Object.Equals(RequestId, Nothing) Or _
                    Object.Equals(RequestId, SampleStoreConstants.EMPTY_STRING) Then
                    RequestId = PayflowUtility.RequestId
                End If

                If Object.Equals(OrigId, Nothing) Or _
                Object.Equals(OrigId, SampleStoreConstants.EMPTY_STRING) Then
                    CreditTrans = New CreditTransaction(UserInfo, _
                                                                    PayflowConnectionData, _
                                                                    Invoice, _
                                                                    Tender, _
                                                                    RequestId)

                Else
                    CreditTrans = New CreditTransaction(OrigId, _
                                                         UserInfo, _
                                                         PayflowConnectionData, _
                                                         Invoice, _
                                                         Tender, _
                                                         RequestId)
                End If

                For Count = 0 To ExtDataList.Count - 1
                    CreditTrans.AddToExtendData(CType(ExtDataList(Count), ExtendData))
                Next

                Dim Verbosity As String = RequestData.Item(SampleStoreConstants.FIELD_VERBOSITY).ToString

                If Not (Object.Equals(Verbosity, Nothing) Or _
                    Verbosity.Length = 0) Then
                    CreditTrans.Verbosity = RequestData.Item(SampleStoreConstants.FIELD_VERBOSITY).ToString
                End If
                CreditTrans.ClientInfo = ClientInfo
                CreditTrans.SubmitTransaction()
                ResponseTable.Add("TRXRESPONSE", CreditTrans.Response())

                Return ResponseTable

            Catch ex As Exception
                Throw New Exception(ex.Message, ex)
            End Try
        End Function
#End Region
    End Class
End Namespace

File: Payflow SDK for .NET/SampleStoreCS/src/PayPal/Payments/SampleStore/CS/Handler/CreditHandler.cs
using System;
using System.Collections;
using PayPal.Payments.DataObjects;
using PayPal.Payments.Transactions;
using PayPal.Payments.SampleStore.CS.Common;
using PayPal.Payments.Common.Utility;



#region "Copyright"

//Unpublished Proprietary Program Material
//This material is proprietary to PayPal, Inc. and is not to be reproduced,
//used or disclosed except in accordance with a written license agreement
//with PayPal, Inc..
//(C) Copyright 2005  PayPal, Inc.   All Rights Reserved.
//PayPal, Inc. believes that this material furnished herewith is accurate
//and reliable.  However, no responsibility, financial or otherwise, can be
// accepted for any consequences arising out of the use of this material.

//The copyright notice above does not evidence any actual or intended
//publication of such source code.

#endregion

namespace PayPal.Payments.SampleStore.CS.Handler
{
	/// -----------------------------------------------------------------------------
	/// <summary>
	/// This class creates object of CreditTransaction class, call its
	/// SubmitTransaction method, gets the response object and returns the same
	/// </summary>
	/// <remarks>
	/// </remarks>
	/// -----------------------------------------------------------------------------
	
	public class CreditHandler : BaseTransactionHandler
	{
		#region "SubmitTransaction"
		/// <summary>
		/// This function overrides the submittransaction function of the base class.
		/// This function creates object of CreditTransaction class, call its
		/// SubmitTransaction method, gets the response object and returns the same
		/// </summary>
		/// <remarks>
		/// </remarks>
		/// -----------------------------------------------------------------------------
		
		public override Hashtable SubmitTransaction(Hashtable RequestData)
		{
			try
			{
				CreditTransaction CreditTrans;
				string mOrigId;
				string mRequestId;
				int mCount;
				Hashtable ResponseTable = new Hashtable();
				
				mRequestId = RequestData[PayPal.Payments.SampleStore.CS.Common.SampleStoreConstants.FIELD_REQUESTID].ToString();
				mOrigId = RequestData[PayPal.Payments.SampleStore.CS.Common.SampleStoreConstants.FIELD_ORIGID].ToString();
				
				if (object.Equals(mRequestId, null) | object.Equals(mRequestId, PayflowConstants.EMPTY_STRING))
				{
					mRequestId = PayflowUtility.RequestId;
				}
				
				if (object.Equals(mOrigId, null) | object.Equals(mOrigId, PayflowConstants.EMPTY_STRING))
				{
					CreditTrans = new CreditTransaction(UserInfo, PayflowConnectionData, Invoice, Tender, mRequestId);
					
				}
				else
				{
					CreditTrans = new CreditTransaction(mOrigId, UserInfo, PayflowConnectionData, Invoice,Tender,mRequestId);
				}
				
				for (mCount = 0; mCount <= ExtDataList.Count - 1; mCount++)
				{
					CreditTrans.AddToExtendData((ExtendData)(ExtDataList[mCount]));
				}

				string mVerbosity = RequestData[PayPal.Payments.SampleStore.CS.Common.SampleStoreConstants.FIELD_VERBOSITY].ToString();
				
				if (!(object.Equals(mVerbosity, null) | mVerbosity.Length == 0))
				{
					CreditTrans.Verbosity = RequestData[PayPal.Payments.SampleStore.CS.Common.SampleStoreConstants.FIELD_VERBOSITY].ToString();
				}
				CreditTrans.ClientInfo = ClientInfo;
				CreditTrans.SubmitTransaction();				
				ResponseTable.Add ("TRXRESPONSE", CreditTrans.Response);
								
				return ResponseTable;
				
			}
			catch (Exception ex)
			{
				throw (new Exception(ex.Message, ex));
			}
		}
		#endregion
	}
}

